river/cluster/denstream.py

```diff
@@ -236,6 +236,15 @@ def _merge(self, point):
                     decaying_factor=self.decaying_factor,
                 )
                 self.o_micro_clusters[len(self.o_micro_clusters)] = mc_from_p
+            merged_status = True
+
+        # when p is not merged and o-micro-cluster set is empty
+        if not merged_status and len(self.o_micro_clusters) == 0:
+            mc_from_p = DenStreamMicroCluster(
+                x=point, timestamp=self.timestamp, decaying_factor=self.decaying_factor
+            )
+            self.o_micro_clusters = {0: mc_from_p}
```

Comment:
Hi @hoanganhngo610, what `self.o_micro_cluster` is before being set here? If it is `None`, calling `len` over it will result in a fail, right?

Reply:
An empty dictionary, just found it. Okay, it will work :D

Reply:
Yep this handles when the length is equal to 0, so basically, an empty dictionary. As I said, the function that queries the closest o-micro-cluster assumes that the set is not empty, so this case is basically left out!